src/Symfony/Component/Form/Form.php

```diff
@@ -1044,11 +1044,29 @@ public function createView(FormView $parent = null)
             $view->children[$name] = $child->createView($view);
         }
 
+        $this->sort($view->children);
+
         $type->finishView($view, $this, $options);
 
         return $view;
     }
 
+    /**
+     * Sorts view's children based on their weight value.
+     */
+    private function sort(array &$children): void
+    {
+        $c = [];
+        $n = 0;
+        foreach ($children as $name => $child) {
+            $c[$name] = ['w' => $child->vars['weight'], 'n' => $n++];
+        }
+
+        uksort($children, static function ($a, $b) use ($c): int {
+            return $c[$a]['w'] === $c[$b]['w'] ? $c[$a]['n'] <=> $c[$b]['n'] : $c[$a]['w'] <=> $c[$b]['w'];
```

Comment:
In PHP < 8.0 the uksort() function doesn't apply a stable sorting to keep equals elements in the original order https://3v4l.org/NQtul. That's why I needed to take care about it here.

Reply:
this can be written as `return [$c[$a]['w'], $c[$a]['n']] <=> [$c[$b]['w'], $c[$b]['n']];` Arrays are sorted by comparing their values one by one

Reply:
Nicer 😍  ! - Updated.